<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1074</width>
    <height>883</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>TasmoEditor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>General</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>MQTT Connection</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <layout class="QGridLayout" name="gridLayout_3">
                <item row="2" column="3">
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>MQTT Password</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>MQTT Username</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="1">
                 <widget class="QPushButton" name="btn_disconnect">
                  <property name="text">
                   <string>Disconnect from MQTT</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QLineEdit" name="txt_pass">
                  <property name="maximumSize">
                   <size>
                    <width>150</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="echoMode">
                   <enum>QLineEdit::PasswordEchoOnEdit</enum>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QPushButton" name="btn_link_device">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Connect to ...</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLineEdit" name="txt_user">
                  <property name="maximumSize">
                   <size>
                    <width>150</width>
                    <height>16777215</height>
                   </size>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>MQTT Ip Address</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="lbl_port">
                  <property name="text">
                   <string>MQTT Port</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="mqtt_port">
                  <property name="maximumSize">
                   <size>
                    <width>150</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>1883</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLineEdit" name="txt_ip">
                  <property name="maximumSize">
                   <size>
                    <width>150</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>192.168.1.100</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="3">
                 <widget class="QComboBox" name="cmb_devices">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item row="5" column="0">
                 <widget class="QPushButton" name="btn_connect_mqtt">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="text">
                   <string>Connect to MQTT</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>Overview</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <layout class="QGridLayout" name="gridLayout_5">
                <item row="1" column="0">
                 <widget class="QListWidget" name="cmd_list_widget">
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>180</width>
                    <height>16777215</height>
                   </size>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QTableWidget" name="CmdtableWidget">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QPushButton" name="btn_refr_list">
                  <property name="text">
                   <string>Get all parameters</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QPushButton" name="btn_show_cmd_docs">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Show Tasmota commands</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QFrame" name="frame_2">
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="1" column="0">
              <widget class="QTextBrowser" name="txt_log">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="btn_clear_log">
               <property name="text">
                <string>Clear Log</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="lbl_log_2">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>50</height>
                </size>
               </property>
               <property name="font">
                <font>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>Log</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="styleSheet">
       <string notr="true"> QProgressBar::chunk
 {
background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(0, 223, 255, 255), stop:1 rgba(255, 255, 255, 255));
 }</string>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="lbl_last_log_txt">
        <property name="maximumSize">
         <size>
          <width>50</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Last Log:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lbl_last_log">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1074</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuTasmHAB">
    <property name="title">
     <string>Menü</string>
    </property>
    <widget class="QMenu" name="menuLoad_Config">
     <property name="title">
      <string>Config</string>
     </property>
     <addaction name="actionLoad_conf"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
    <addaction name="menuLoad_Config"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionInfo"/>
   </widget>
   <addaction name="menuTasmHAB"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_save_conf">
   <property name="text">
    <string>Save Device Config</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionInfo">
   <property name="text">
    <string>Info</string>
   </property>
  </action>
  <action name="actionLoad_conf">
   <property name="text">
    <string>Load Config</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>txt_ip</tabstop>
  <tabstop>txt_user</tabstop>
  <tabstop>txt_pass</tabstop>
  <tabstop>tabWidget</tabstop>
 </tabstops>
 <resources>
  <include location="resource.qrc"/>
  <include location="../TasmoHAB/resource.qrc"/>
 </resources>
 <connections/>
</ui>
